feature: render highlight color in article note

aaachen opened the pull request:
## Issue

Issue: https://github.com/omnivore-app/obsidian-omnivore/issues/166

feature: render highlight color in article note
improvement: refactor settings tab into its own file

## Description

Two options here for highlight css
1. directly configure color mapping in Omnivore plugin, comes with rounded highlight but user can choose to customize in their css
1. delegate color mapping to Highlightr plugin (by setting corresponding highlight css class). Benefit of this option is more highlight style the plugin supports, con is it's another dependency
- Ensure to pick the `css-class` style option in Highlightr plugin 

### Caveat

- Some themes (and plugins like Highlightr) may have global css setting that overwrite the css the omnivore plugin sets, be aware to customize around id. One such theme I've tested with is Prism

## Testing
- Unit Test ✅ -  PASS  src/__tests__/renderHighlightColorQuote.spec.ts (5.667 s)
- Manual Test ✅ - See screenshots below

## Screenshots

![settings-1](https://github.com/omnivore-app/obsidian-omnivore/assets/7253271/1105f95c-e468-4318-aa9e-cd788a84669b)

- Default setting does not render highlight color

![settings-2](https://github.com/omnivore-app/obsidian-omnivore/assets/7253271/abb1ffab-27c1-4ca7-ba0c-873e09da4f59)

- Settings (configure color mapping in omnivore plugin)

![settings-3(highlightr)](https://github.com/omnivore-app/obsidian-omnivore/assets/7253271/71037b42-b42d-46ba-9e3b-04319bad6b6e)

- Settings (configure color mapping in highlightr plugin)

![article-1](https://github.com/omnivore-app/obsidian-omnivore/assets/7253271/44527b1b-8556-4e38-bcd7-ce7a916a19d9)

- Article page

![article-2](https://github.com/omnivore-app/obsidian-omnivore/assets/7253271/ca2787b1-c6ea-44ac-943e-f5d1cf163978)

- Reference

![highlightr-style](https://github.com/omnivore-app/obsidian-omnivore/assets/7253271/bec79512-f3f4-4f6b-bd00-a819d6261dfb)

- Using style of Highlightr plugin

![settings-overview](https://github.com/omnivore-app/obsidian-omnivore/assets/7253271/581d5d78-5b95-4dc8-9927-def1ebaed3d1)

- Refactored settings page with section title grouping together the relevant information

sywhb reviewed (approved):
Look great to me. Really appreciate it as this is one of the biggest request from the users

sywhb commented:
@aaachen Sorry I think the `develop` branch is a bit outdated due to some of the latest work pushed to `master` branch so I rebased the branch and it created some conflicts 😢

aaachen commented:
> @aaachen Sorry I think the `develop` branch is a bit outdated due to some of the latest work pushed to `master` branch so I rebased the branch and it created some conflicts 😢

@sywhb Fixed the conflicts in respect to master, lmk if you see any problem

p.s. also removed the "Settings for Omnivore Plugin" header according to [the guideline here](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Only+use+headings+under+settings+if+you+have+more+than+one+section.)
